packages/vitest/src/runtime/run.ts

```diff
@@ -27,6 +27,20 @@ export async function runTest(test: Test) {
   if (test.mode !== 'run')
     return
 
+  const regex = new RegExp(process.__vitest_worker__.ctx.config.testNamePattern)
```

Comment:
Could we create the regex on configuring resolving instead of every test running?